### PR TITLE
[NCL-7595] Implement SCM Repository usages list

### PR DESCRIPTION
**Pull request by DnsZhou**

### CODE REVIEW

File: src/services/scmRepositoryApi.ts [hunk 1]
```diff
@@ -53,3 +55,28 @@ export const patchScmRepository = (
 ) => {
   return pncClient.getHttpClient().patch<SCMRepository>(`/scm-repositories/${id}`, patchData, requestConfig);
 };
+
+interface IGetByScmRepositoryData {
+  scmRepositoryId?: string;
```
Comment (patrikk0123) [inline src/services/scmRepositoryApi.ts:60]: Should be mandatory, I presume.

File: src/services/scmRepositoryApi.ts [hunk 2]
```diff
@@ -53,3 +55,28 @@ export const patchScmRepository = (
...
+}
+
+/**
+ * Get all Build Configs of a SCM Repository with the latest Build.
+ *
+ * @param scmRepositoryId - The id of the SCM Repository
+ *
```
Comment (patrikk0123) [inline src/services/scmRepositoryApi.ts:67]: Remove empty line.

File: src/services/scmRepositoryApi.ts [hunk 3]
```diff
@@ -53,3 +55,28 @@ export const patchScmRepository = (
...
 };
+
+interface IGetByScmRepositoryData {
+  scmRepositoryId?: string;
+}
+
+/**
+ * Get all Build Configs of a SCM Repository with the latest Build.
+ *
+ * @param scmRepositoryId - The id of the SCM Repository
```
Comment (patrikk0123) [inline src/services/scmRepositoryApi.ts:66]: `scmRepositoryId - SCM Repository ID`

File: src/services/scmRepositoryApi.ts [hunk 4]
```diff
@@ -53,3 +55,28 @@ export const patchScmRepository = (
...
+ *
+ * @param requestConfig - Axios based request config
+ */
+export const getBuildConfigsWithLatestBuild = (
+  { scmRepositoryId }: IGetByScmRepositoryData,
+  requestConfig: AxiosRequestConfig = {}
+) => {
+  const qParam = addQParamItem(
+    'scmRepository.id',
+    scmRepositoryId!,
```
Comment (patrikk0123) [inline src/services/scmRepositoryApi.ts:76]: Remove !

File: src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx [hunk 5]
```diff
@@ -82,9 +95,7 @@ export const ScmRepositoryDetailPage = () => {
           </ToolbarItem>
         </Toolbar>
         <ContentBox borderTop>
-          <div style={{ width: '100%', height: '30vh', textAlign: 'center', paddingTop: '30px' }}>
-            TODO: Add Usages table here and remove the placeholder
-          </div>
+          <BuildConfigsList {...{ serviceContainerBuildConfigs: serviceContainerBuildConfigs, componentId }} />
```
Comment (patrikk0123) [inline src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx:98]: `serviceContainerBuildConfigs: serviceContainerBuildConfigs` => `serviceContainerBuildConfigs`

File: src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx [hunk 6]
```diff
@@ -22,16 +24,27 @@ import * as scmRepositoryApi from 'services/scmRepositoryApi';
 import { generateScmRepositoryName } from 'utils/entityNameGenerators';
 import { generatePageTitle } from 'utils/titleHelper';
 
-export const ScmRepositoryDetailPage = () => {
+interface IScmRepositoryDetailPageProps {
+  componentId?: string;
+}
+
+export const ScmRepositoryDetailPage = ({ componentId = 's2' }: IScmRepositoryDetailPageProps) => {
```
Comment (patrikk0123) [inline src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx:31]: `s1`

**Review by patrikk0123 — COMMENTED** [review]
Also, please, rename the commit message (usages typo).

File: src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx (hunk 6 above)
Comment (DnsZhou) [reply src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx:31]: we already have `ScmRepositoriesPage = ({ componentId = 's1' }`,  I would like to name it differently.

File: src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx (hunk 5 above)
Comment (DnsZhou) [reply src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx:98]: Do you mean `<BuildConfigsList {...{ serviceContainerBuildConfigs, componentId }} />` instead?

Comment (patrikk0123) [reply src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx]: yes, indeed

File: src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx (hunk 6 above)
Comment (patrikk0123) [reply src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx:31]: It does not matter as long as the pages are different, but okay. The only important thing is that component IDs are distinct and used consistently on one page. Other than that, the name is inconsequential.

**Review by patrikk0123 — APPROVED** [review]

**Pull request merged**